### ISSUE TITLE
method signatures

### ISSUE THREAD

**Issue by ghost** [datrs/varinteger#5]
`encode()` and `decode()` both have failure modes if you pass in a source or destination buffer that is too small but the return type is `usize` for both `encode()` and `decode()`.

The `_with_offset()` routines could likewise be removed, since slices in rust are very cheap.

These are the routines I'm using inline in desert but I considered using this varinteger first:

https://github.com/substack/desert/blob/46789f4af0e98afb80438cb80db573d0d8dbb630/src/lib.rs#L1020-L1054

**Comment by okdistribute** [comment]
Also ran into this issue trying to use this package. If I put together a pull request, would someone be able to review & publish any relevant changes? (also would be nice to put out feelers for who is actively maintaining this repo!)  @yoshuawuyts @Frando  @khernyo

**Comment by Frando** [comment]
Yup, stumbled over this too and agree. The signatures of the methods @substack posted look good to me! I would review a PR. I don't know if someone actively maintains this currently? If not once we're confident with a 2.0 version (the signature change to return a `Result` is breaking) I'll look to get it published (there's a few people with publish right to datrs modules, including @bltavares and me) or someone steps up until then I'd say. :-)

**Comment by okdistribute** [comment]
Thanks @Frando ! I'll work up a PR soon.

**Comment by okdistribute** [comment]
Hey, do you have any thoughts on this [PR](https://github.com/datrs/varinteger/pull/6) @Frando @substack? Thanks for taking a look!
